src/ast/passes/codegen_llvm.cpp

```diff
@@ -756,18 +756,22 @@ void CodegenLLVM::visit(Call &call)
     b_.CreateLifetimeEnd(key);
     expr_ = nullptr;
   } else if (call.func == "delete") {
-    for (const auto &arg : call.vargs) {
-      auto &map = static_cast<Map &>(*arg);
-      auto [key, scoped_key_deleter] = getMapKey(map);
-      if (!is_bpf_map_clearable(map_types_[map.ident])) {
-        // store zero instead of calling bpf_map_delete_elem()
-        AllocaInst *val = b_.CreateAllocaBPF(map.type, map.ident + "_zero");
-        b_.CreateStore(Constant::getNullValue(b_.GetType(map.type)), val);
-        b_.CreateMapUpdateElem(ctx_, map.ident, key, val, call.loc);
-        b_.CreateLifetimeEnd(val);
-      } else {
-        b_.CreateMapDeleteElem(ctx_, map, key, call.loc);
-      }
+    auto &arg0 = *call.vargs.at(0);
+    auto &map = static_cast<Map &>(arg0);
+    // Note: delete accepts two arguments except in the case of scalar maps
+    // (maps with no keys) in which case it you can just pass it the map
+    // and it will act similar to `clear` e.g. `delete(@scalar);`
```

Comment:
Also supports one argument for the legacy delete syntax!
`delete(@map[123])`

Reply:
Haha fine. But that's going away.

Reply:
I'm not sure it should go away - it's been part of the language for so long that all existing scripts are using it. Should we really break them?

Reply:
We can create a separate issue to discuss but my 2cents - yes we should deprecate and eventually remove. From someone else reading the script the syntax makes no sense. `delete(@map[123])` looks like we're deleting the value and deleting it from what? Also do we want to set a precedence that we can't delete anything from the bpftrace API? We could do this in phases e.g. begin by one release issuing a warning then the next release it gets deleted.

Reply:
Either way, I updated the comment language in the codegen file.